### PR TITLE
Feature/extra interfaces

### PR DESCRIPTION
**Pull request by Klynger**
#### What is the purpose of this pull request?
Create the extra interface to make it possible to use the components as slots

#### How should this be manually tested?

https://newmenu--storecomponents.myvtex.com/

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/SubmenuItemHorizontalList.tsx [hunk 1]
```diff
@@ -29,9 +29,8 @@ export default function SubmenuItemHorizontalList(props: Props) {
     TopItem = MenuItemDefault,
     ListItem = MenuListItemDefault,
   } = props
-  const subNavigation = navigationItem?.subNavigation
+  const navigationId = navigationItem?.subNavigation ?? id
```
Comment (victorhmp) [inline react/SubmenuItemHorizontalList.tsx:32]: Could `id` be `undefined`?
And would `useNavigation` know how to handle that?

File: store/interfaces.json [hunk 2]
```diff
@@ -2,5 +2,25 @@
...
+    "composition": "children"
+  },
+  "menu-item.vertical-list": {
+    "component": "SubmenuItemVerticalList",
+    "composition": "children"
+  },
+  "menu-item.horizontal-list": {
+    "component": "SubmenuItemHorizontalList",
+    "composition": "children"
+  },
```
Comment (victorhmp) [inline store/interfaces.json:21]: How about changing the name on these components?
`menu-item.vertical-list` rendering a component `SubmenuItemVerticalList` might be confusing.

File: react/SubmenuItemHorizontalList.tsx (hunk 1 above)
Comment (Klynger) [reply react/SubmenuItemHorizontalList.tsx:32]: if `navigationId` gets to be `null` `useNavigation` will return `null`. But id should never be undefined

File: react/MenuItemVerticalList.tsx [hunk 3]
```diff
@@ -21,14 +21,15 @@ const CSS_HANDLES = [
...
+export default function MenuItemVerticalList(props: Props) {
   const {
+    id,
     navigationItem,
     TopItem = MenuItemDefault,
     ListItem = MenuItemDefault,
     linkItemClasses,
   } = props
-  const subNavigation = navigationItem?.subNavigation
+  const subNavigation = navigationItem?.subNavigation ?? id
```
Comment (iaronaraujo) [inline react/MenuItemVerticalList.tsx:32]: In which situations will the `id` be used?

File: react/hooks/useNavigation.ts [hunk 4]
```diff
@@ -4,138 +4,498 @@ import { NavigationId, Navigation } from 'navigation'
 // real data
 const navigationsMock: Record<NavigationId, Navigation> = {
```
Comment (iaronaraujo) [inline react/hooks/useNavigation.ts:5]: Is this test data? If so, shouldn't this be on the test folder?

**Pull request merged**